src/runtime/GlobalObjectBuiltinDataView.cpp

```diff
@@ -83,6 +79,10 @@ Value builtinDataViewConstructor(ExecutionState& state, Value thisValue, size_t
     ArrayBufferView* obj = new DataViewObject(state, proto);
     obj->setBuffer(buffer, byteOffset, byteLength);
 
+    if (obj->buffer()->isDetachedBuffer()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().DataView.string(), false, String::emptyString, "%s: ArrayBuffer is detached buffer");
```

Comment:
We can use predefined error messages here..
For this case `ErrorObject::Messages::GlobalObject_DetachedBuffer` would be suitable.